runner/local/simple.go

```diff
@@ -85,52 +99,64 @@ func (r *simpleRunner) Erase(run runner.RunId) {
 	}
 }
 
-func (r *simpleRunner) abortImpl(run runner.RunId, state runner.ProcessState) runner.ProcessStatus {
+func (r *simpleRunner) updateStatus(new runner.ProcessStatus) runner.ProcessStatus {
 	r.mu.Lock()
 	defer r.mu.Unlock()
-	result, ok := r.running[run]
+	old, ok := r.runs[new.RunId]
 	if !ok {
-		return runner.BadRequestStatus(run, fmt.Errorf("could not find abortable run: %v", run))
+		return runner.BadRequestStatus(new.RunId, fmt.Errorf("cannot find run %v", new.RunId))
+	}
+	if old.State.IsDone() {
+		return old
 	}
-	result.Abort()
-	delete(r.running, run)
-	r.runs[run] = runner.AbortStatus(run)
-	return r.runs[run]
+	r.runs[new.RunId] = new
+	if new.State.IsDone() && r.running.runId == new.RunId {
```

Comment:
This is a nit regarding naming then, because I wouldn't have given a second thought to closing a 'doneCh'. It doesn't really make sense to rename it though.